axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveServiceImpl.java

```diff
@@ -900,8 +904,149 @@ public TaxLine getTaxLine(Move move, MoveLine moveLine, Account accountingAccoun
 
   @Override
   @Transactional
-  public void generateCounterpartMoveLine(Move move) {
-    move.addMoveLineListItem(createCounterpartMoveLine(move));
+  public void generateCounterpartMoveLine(Move move) throws AxelorException {
+    MoveLine counterPartMoveLine = createCounterpartMoveLine(move);
+    move.addMoveLineListItem(counterPartMoveLine);
+    generateCounterpartAnalyticMoveLine(move, counterPartMoveLine);
     moveRepository.save(move);
   }
+
+  public void generateCounterpartAnalyticMoveLine(Move move, MoveLine counterpartMoveLine)
+      throws AxelorException {
+    List<AnalyticAccount> analyticAccounts = getAnalyticAccountList(move);
+    List<AnalyticMoveLine> analyticMoveLines = getAnalyticMoveLines(move);
+    counterpartMoveLine.clearAnalyticMoveLineList();
+    if (appAccountService.getAppAccount().getManageAnalyticAccounting()
+        && move.getCompany().getAccountConfig().getManageAnalyticAccounting()
+        && hasToCreateCounterpartAnalytics(move)) {
+      for (AnalyticAccount analyticAccount : analyticAccounts) {
+        counterpartMoveLine.addAnalyticMoveLineListItem(
+            createAnalyticMoveLine(analyticAccount, analyticMoveLines, counterpartMoveLine));
+      }
+
+      if (counterpartMoveLine.getAnalyticMoveLineList() != null) {
+        counterpartMoveLine =
+            computeCounterpartAnalyticMoveLines(counterpartMoveLine, analyticMoveLines);
+      }
+    }
+  }
+
+  protected boolean hasToCreateCounterpartAnalytics(Move move) {
+    boolean result = true;
+    int analyticMoveLineAmount = 0;
+    if (!(move.getJournal()
+            .getJournalType()
```

Comment:
journalType is not required so a NPE can be thrown here, please change the method to manage this case.